qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -168,5 +204,181 @@ class DefaultSynthesisLinearFunction(HighLevelSynthesisPlugin):
 
     def run(self, high_level_object, **options):
         """Run synthesis for the given LinearFunction."""
-        decomposition = synth_cnot_count_full_pmh(high_level_object.linear)
+        # For now, use PMH algorithm by default
+        decomposition = PMHSynthesisLinearFunction().run(high_level_object, **options)
         return decomposition
+
+
+class KMSSynthesisLinearFunction(HighLevelSynthesisPlugin):
+    """Linear function synthesis plugin based on the Kutin-Moulton-Smithline method."""
+
+    def __init__(self, **options):
+        self._options = options
+
+    def run(self, high_level_object, **options):
+        """Run synthesis for the given LinearFunction."""
+
+        # Combine the options passed in the initializer and now,
+        # prioritizing values passed now.
+        run_options = self._options.copy()
+        run_options.update(options)
+
+        # options supported by this plugin
```

Comment:
could you please add some explanation or documentation on these options?

Reply:
thanks, done in [a87d495](https://github.com/Qiskit/qiskit-terra/pull/9250/commits/a87d495bb08b05109652663dcb9d247d0d9d7aa0)